tests/ut_screenlock/ut_screenlock.cpp

```diff
@@ -140,22 +117,28 @@ void Ut_ScreenLock::testUnlockScreenWhenLocked()
     screenLock->setEventEaterEnabled(true);
     screenLock->unlockScreen();
 
-    QCOMPARE(qDbusAbstractInterfaceCallMethod, TEST_METHOD);
-    QCOMPARE(qDbusAbstractInterfaceCallPath, TEST_PATH);
-    QCOMPARE(qDbusAbstractInterfaceCallService, TEST_SERVICE);
-    QCOMPARE(qDbusAbstractInterfaceCallInterface, TEST_INTERFACE);
-    QCOMPARE(qDbusAbstractInterfaceCallMode, QDBus::NoBlock);
-    QCOMPARE(qDbusAbstractInterfaceCallArg1.toInt(), (int)ScreenLock::TkLockUnlock);
+    QString member(screenLock->m_callbackMethod.member());
+    QString path(screenLock->m_callbackMethod.path());
+    QString service(screenLock->m_callbackMethod.service());
+    QString interface(screenLock->m_callbackMethod.interface());
+    QList<QVariant> arguments(screenLock->m_callbackMethod.arguments());
+    bool modeOk = false;
+    int mode(arguments.value(0).toInt(&modeOk));
+    if (!modeOk) {
+        qWarning("mode arg not present");
+        mode = -1;
+    }
+
+    QCOMPARE(member, TEST_METHOD);
+    QCOMPARE(path, TEST_PATH);
+    QCOMPARE(service, TEST_SERVICE);
+    QCOMPARE(interface, TEST_INTERFACE);
+    QCOMPARE(mode, int(ScreenLock::TkLockUnlock));
 }
 
 void Ut_ScreenLock::testUnlockScreenWhenNotLocked()
 {
     screenLock->unlockScreen();
```

Comment:
This is not really testing anything anymore? Could delete?

Reply:
Maybe. then again I'm not sure these tests are context free i.e. leaving this out just might break the following tests.